src/Adapter/PhpredisAdapter.php

```diff
@@ -55,6 +55,10 @@ public function getMulti(array $keys): array
         $result = $this->getClient()->mGet($keys);
         $values = [];
 
+        if (!is_array($result)) {
```

Comment:
Why return an empty array? This can cause confusion. If the key contains any value `!Is_array` it should return an exception saying that the returned value is not an array, instead of returning an empty array, which can cause confusion in the implementation of this method.

Low-level methods must always avoid returning "default" values, unless the value comes from the cache, otherwise it should return an exception.

What you think? Maybe I'm wrong, but that's what I think about it.

Reply:
And also do we have tests for this scenario?

Reply:
@juniorb2ss the thing is that cache is returning for some occasions a `bool` value, for this y idea was to verify if the actual value was an array, if not return empty values. What do you think? Or is better to handle this as an exception even if the value comes from cache?

Reply:
Well then if the value comes from the cache, as @juniorb2ss mentioned before, this "default" value is acceptable, maybe add a comment (not my favorite practice) documenting this? what do you think guys?

Reply:
I don't like to return default values in low-level libraries, this can cause confusion in the implementation. An example:

The key does not exist: returns empty array
The key exists, but it contains empty array

If the application accepts empty array but has rules for when the rule does not exist, that way it will never know, it will only receive empty array.

Low-level libraries, in my view, cannot have this abstraction, they need to be clear.

Key does not exist? Exception, tells the application what is really happening, does not hide the problem.

Reply:
If the value returned to MGET is different from the expected value, then another exception should be thrown, informing which value was received and which was the expected value.

or, let the application handle it and return a mixed value (but with documentation telling this)

Reply:
@josuegpuig if you return an exception here as @juniorb2ss suggest, that exception should be managed by `shop-front`. After reading the last 2 comments and thinking about avoiding comments(complex documentation) in the code, I agree with @juniorb2ss suggestion but please, just be sure to manage this correctly on the other side. 

cc @tamara-dietz

Reply:
Whatever you do, it should generally follow https://github.com/phpredis/phpredis/issues/1810 which is the upstream issue for this.

Reply:
@nesl247 thank you! With this I think is better to handle it by the app.

Reply:
@juniorb2ss @smmd thank you for your feedback, I will manage this in `shop-front` and only return here if necessary